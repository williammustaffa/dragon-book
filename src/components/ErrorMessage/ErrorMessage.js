import React from "react";
import PropTypes from "prop-types";
import { Message } from "semantic-ui-react";

function ErrorMessage({ title, content }) {
  return (
    <Message negative>
      {title && <Message.Header>{title}</Message.Header>}
      <p>{content}</p>
    </Message>
  );
}

ErrorMessage.propTypes = {
  title: PropTypes.string,
  content: PropTypes.string.isRequired,
};

export default ErrorMessage;