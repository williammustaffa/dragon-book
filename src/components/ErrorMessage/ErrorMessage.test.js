import React from "react";
import ErrorMessage from "./ErrorMessage";
import { render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"

test("Test ErrorMessage component", () => {
  const { asFragment } = render(
    <ErrorMessage
      title="Sample title"
      content="Sample error"
    />
  );

  // Run tests
  expect(asFragment()).toMatchSnapshot();
});