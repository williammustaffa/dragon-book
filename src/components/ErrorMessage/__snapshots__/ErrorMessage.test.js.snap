// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ErrorMessage component 1`] = `
<DocumentFragment>
  <div
    class="ui negative message"
  >
    <div
      class="header"
    >
      Sample title
    </div>
    <p>
      Sample error
    </p>
  </div>
</DocumentFragment>
`;
